go/marshal/encode.go

```diff
@@ -260,15 +274,24 @@ func nomsType(t reflect.Type) *types.Type {
 	case reflect.String:
 		return types.StringType
 	case reflect.Struct:
-		return structNomsType(t)
-	default:
-		// This will be reported as an error at a different layer.
-		return nil
+		for i, pst := range parentStructTypes {
```

Comment:
I would suggest that this makes more sense in structNomsType(), but ... it's a gentle suggestion.